lib/bokken_web/controllers/reset_password_controller.ex

```diff
@@ -0,0 +1,70 @@
+defmodule BokkenWeb.ResetPasswordController do
+  use BokkenWeb, :controller
+
+  alias Bokken.Accounts
+  alias Bokken.Accounts.User
+  alias Bokken.Mailer
+  alias Bokken.Repo
+  alias BokkenWeb.AuthEmails
+
+  def create(conn, %{"user" => %{"email" => email}}) do
+    case Accounts.get_user_by_email(email) do
+      nil ->
+        conn
+        |> put_status(:not_found)
+        |> render("error.json", error: "There is no such user with that email.")
+
+      user ->
+        user = Accounts.reset_password_token(user)
+
+        AuthEmails.reset_password_email(user.reset_password_token, to: user.email)
+        |> Mailer.deliver()
+
+        conn
+        |> put_status(:created)
+        |> render("show.json", %{})
+    end
+  end
+
+  def update(conn, %{"id" => token, "user" => params}) do
```

Comment:
You should refactor some of this into the Accounts context. Also, everything should be done in one transaction only, as to avoid cases where the first steps work but the later fail.